### ISSUE TITLE
Duplicate Quality Audit entries cause inaccurate Donut

### ISSUE THREAD

**Issue by pjw6193** [pjw6193/caliber#558]
### Issue Description
The Reports page shows inaccurate data on the QC donut. In a batch of 16 people, for example, the donut shows 22 QC grades for a week. This issue was caused by duplicated entries in the NOTES table. 

### Expected behavior
Each smiley in a Quality Audit is recorded with value of 1 and displayed accordingly for that week

### Actual behavior (bugs)
Report not lining up with the number of smileys entered. Somehow during the audit in question, Caliber inserted new records into NOTES instead of updating the existing notes when smileys were selected and/or new note content was entered. 

### Steps for replicating (bugs)
On Quality Audit page, click smileys for everyone. Then, access Reports and select a batch and a week.

### Screenshots (if applicable)

Report table
![qc-status-count](https://user-images.githubusercontent.com/20826349/27978817-c09387ca-633f-11e7-87d5-f08db2580145.png)

Quality Audit for the week (Reports)
![qc-status-count-2](https://user-images.githubusercontent.com/20826349/27978813-b79cd158-633f-11e7-99e3-7479fa9f64de.png)

Ran this script in prod:
select * from caliber_note where TRAINEE_ID in (SELECT trainee.trainee_id 
FROM CALIBER_BATCH 
b inner join caliber_trainee trainee on trainee.batch_id = b.batch_id 
where b.batch_id = 112500) and IS_QC_FEEDBACK = 1 and WEEK_NUMBER = 1;

and got these results:

![qc-status-count-db](https://user-images.githubusercontent.com/20826349/27978826-d89c633c-633f-11e7-9986-12aef1bcf568.png)

**Comment by pjw6193** [comment]
At one point in v1.1, QC Assess team had this duplicate record issue when you would type the notes first then select the smiley. I told them it shouldn't matter whether you select the smiley first or add notes first. The $http call should update the same record. I guess some edge case slipped thru.. maybe clicking very quickly before Hibernate can persist the first call to the db??

**Comment by pjw6193** [comment]
manually deleted duplicates in the DB to get PROD back on track, but we need to isolate the root cause here

**Comment by pjw6193** [comment]
The following query will identify the duplicate records for QC_TRAINEE notes each week:

```
select qc_status, count(QC_STATUS), caliber_trainee.trainee_name, week_number
from caliber_note
inner join caliber_trainee
on caliber_trainee.TRAINEE_ID = caliber_note.TRAINEE_ID
where note_type = 'QC_TRAINEE'
group by QC_STATUS, caliber_trainee.trainee_name, week_number
having count(QC_STATUS) > 1
order by caliber_trainee.trainee_name;
```

Still troubleshooting this issue.

**Comment by pjw6193** [comment]
closed by PR #594
